Added alternative text for loading spinner

iamtomhanks opened the pull request:
<!--- Provide a general summary of your changes in the title above -->

## Description
<!--- Describe your changes in detail -->
For browsers that don't support CSS animations, the user will see 'Loading...' instead of the loader. 

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
https://trello.com/c/zzk6QH6Y/247-loading-spinner-needs-a-text-alternative

## How has this been tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Tested locally, and if I explicitly return false from the isCSSAnimationsSupported function, the loading text is shown. However, since CSS animations have been supported since Chrome version 43, I tried testing on Chrome version 42 and it looks like we don't support that in general.

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/18446243/53370439-b67c2400-391b-11e9-936d-bee7957d7df6.png)

## Types of changes
<!--- What types of changes does your code introduce? -->
<!--- Please remove all lines which don't apply. -->
- ✅ New feature (non-breaking change which adds functionality)

## Checklist:
<!--- Please go over all the following points. -->
<!--- Again, remove any lines which don't apply. -->
<!--- Pull Requests that don't fulfill all [REQUIRED] requisites are likely -->
<!--- to be sent back to you for correction or will be rejected.  -->
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

iamtomhanks commented:
Is there any other case for which text should be shown instead of the loader? Besides when CSS animations are not supported?

iamtomhanks commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Loading spinner needs a text alternative](https://trello.com/c/zzk6QH6Y/247-loading-spinner-needs-a-text-alternative)

mlippert commented:
rebased to head of develop (resolved conflict in App.js).
Waiting for @arizzitano 's review before merging.

arizzitano commented on src/components/App.js:
```diff
@@ -58,11 +58,24 @@ const SkipContent = styled.div.attrs({
 const LoadingView = () => {
   return (
     <div className="columns has-text-centered is-centered is-vcentered" style={{minHeight: "100vh"}}>
       <div className="column is-vcentered has-text-centered">
```
```suggestion
      <div className="column is-vcentered has-text-centered" ariaLabel="Loading">
```

mlippert commented on src/libs/utils.js:
```diff
@@ -34,3 +34,35 @@ export function isScreenShareSourceAvailable() {
+  let animation = false,
+      animationstring = 'animation',
+      keyframeprefix = '',
+      domPrefixes = 'Webkit Moz O ms Khtml'.split(' '),
```
why is this better than:
```
domPrefixes = [ 'Webkit', 'Moz', 'O', 'ms', 'Khtml' ],
```
it doesn't seem clearer, and it's additional processing to split the string and create the array every time the function is called. It would make more sense to me to use split if the string came from somewhere else. Does the string live in that form in some other location?

mlippert commented:
@iamtomhanks when you've addressed the comments (and looked at the style changes in the commit I added. you can see the commit comments by clicking on the **...** next to the commit above) in a new commit in this branch (no rebasing or squashing needed), I'll check it out again and get it merged. Thanks.

iamtomhanks commented:
@mlippert Sorry, that was a code snippet found online. I'll be more careful with the styling in the future, those are styles that I wouldn't normally use. That split function does seem weird to me... not sure what the motivation behind that would have been. I've corrected that.

mlippert commented:
@iamtomhanks Thank you for addressing my comment about using the `split` function.

However, you don't seem to have really _looked at_ the comments in this PR. In the commit I added was a request for you to fill in some additional information, which is what makes me think you didn't look at what was actually changed by that commit?

What I was hoping you'd add (replacing the comment `(Brec please fill in the logic used because it isn't immediately obvious -mjl)`), now that you've given me the link above, was something like:

```
/**
 * Determine if CSS animations are supported
 *
 * This function was found at:
 * @see <https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Animations/Detecting_CSS_animation_support>
 *
 * Here's the explanation of how this works (from the above page):
 *  For starters we define a few variables. We assume that animation is not
 *  supported by setting animation to false. We set the animationstring to
 *  animation which is the property we want to set later on. We create an array
 *  of browser prefixes to loop over and we set pfx to an empty string.
 *
 *  Then we check if the CSS animation-name property on the style collection for
 *  the element specified by the variable elem is set. This means the browser
 *  supports CSS animation without any prefix, which, to date, none of them do.
 *
 *  If the browser does not support non-prefixed animation and animation is still
 *  false, we iterate over all the possible prefixes, since all the major
 *  browsers are currently prefixing this property and changing its name to
 *  AnimationName instead.
 *
 *  Once this code is finished running, the value of animation will be false if
 *  CSS animation support isn't available, or it will be true. If it is true then
 *  both the animation property name and the keyframe prefix will be the right
 *  ones. So if you use a new Firefox, the property will be MozAnimation and the
 *  keyframe prefix -moz- and with Chrome it'll be WebkitAnimation and -webkit-.
 *  Notice browsers don't make it easy with the switching between camelCase and
 *  hyphen-ation.
 */
``` 
It's always good to leave an attribution of where complicated code snippets came from in the code.

Ari also suggested some changes you haven't addressed.

iamtomhanks commented:
@mlippert Sorry about that, I didn't see that comment. It does make sense to provide documentation and/or a resource link for third-party code snippets. I think I've added that correctly and addressed Ari's comment in a new commit.
